clients/web/src/collections/Collection.js

```diff
@@ -54,12 +74,29 @@ var Collection = Backbone.Collection.extend({
      * Fetch the previous page of this collection, emitting g:changed when done.
      */
     fetchPreviousPage: function (params) {
-        this.offset = Math.max(0, this.offset - this.length - this.pageLimit);
+        if (this.filterFunc) {
+            if (this.append) {
+                this.offset = 0;
+            } else {
+                this.pageOffsetStack = this.pageOffsetStack || [];
+                if (this.pageOffsetStack.length) {
+                    this.pageOffsetStack.pop();
+                }
+
+                this.offset = (
+                    this.pageOffsetStack.length ? this.pageOffsetStack.pop() : 0
```

Comment:
That's because the fetch ends up pushing the offset back on.  So, the second pop is more of a peek.